src/java/org/apache/cassandra/schema/Schema.java

```diff
@@ -92,6 +93,8 @@ public class Schema implements SchemaProvider
     public static final String FORCE_LOAD_LOCAL_KEYSPACES_PROP = "cassandra.schema.force_load_local_keyspaces";
     private static final boolean FORCE_LOAD_LOCAL_KEYSPACES = Boolean.getBoolean(FORCE_LOAD_LOCAL_KEYSPACES_PROP);
 
+    private static final Set<String> TEMPORARILY_UNMAPPED_SYSTEM_TABLE = Sets.newConcurrentHashSet();
```

Comment:
i'm a bit confused by this concurrency approach.

it's only using two values, and they are both mapped and unmapped at the same time. why not just use an atomic boolean ? 

```
private static final AtomicBoolean TEMPORARILY_UNMAPPED_SYSTEM_TABLES = new AtomicBoolean(…);
```

And i'm a bit confused about the concurrency here:  if this can be called concurrently to CassandraDaemon.setup(), then could it be called before we even got to call ` unmapSystemTable(…)`

e.g. as opposed to just having an initialised boolean that gets set at the end of `convertToNodesFilesIfNecessary()`

Reply:
Using a boolean flag would make all system tables unmapped while I only want to unmap two (`local` and `peers`).

Regarding concurrency: I was thinking about whether any operation could read the tables before the node finishes starting up and I came to the confusion it isn't going to happen. What I am worried about is the visibility hence the concurrent set (happy to be corrected if I am overeager).

Reply:
yeah, i was thinking you'd just hardcode a boolean flag for those two system tables…
appreciate that's not entirely clean, to bring that logic into here, but the flag/map concurrency construct is doing that regardless… 

entirely trivial, but at least from what i've understood so far, encapsulation and readability seems cleaner if line 472 was 
```
if (NodeConstants.canBeMapped(keyspaceName, tableName) && LegacySystemKeyspaceToNodes.isInitialized())
```
🤷

Reply:
yes, that.

Reply:
I gave it a try but wasn't too happy with the result, the tests needed to fake the non-initialized state to read the unmapped tables which felt dirty. That's why I appreciated the additional map/unmap layer that code in `Schema` class provides.

Reply:
> I gave it a try but wasn't too happy with the result, the tests needed to fake the non-initialized state to read the unmapped tables which felt dirty.

I must be missing something, isn't that just to un-initialise it (regardless of where it is) ?

Reply:
for example with this change:
 https://github.com/datastax/cassandra/compare/DSP-23956...thelastpickle:cassandra:mck/DSP-23956 

 the test can just call 
```
LegacySystemKeyspaceToNodes.initialized.set(false);
```

Reply:
Yes, it is and to me, it looks less clear when reading the test.

Reply:
Fixed.